### PR TITLE
Trireme 8.x

### PR DESCRIPTION
**Pull request by dstiliadis**
This PR will update the example to trireme-lib 8.x 

Some significant addition is the example policy where we demonstrate how mutual authorization can be used to decouple policies. Details have been added in the Readme. 

Essentially Trireme enables different policies to be applied by developers and operators:
1. Developers define the services they consume (and thus, their service depends on).
2. Operators define the rules of exposing services to different entities.

The mutual authorization automatically guarantees that the policies both operators and developers are met, while minimizing any interaction that other entities can have with the system. 
Developers can declare their dependencies as part of their code commit.

### CODE REVIEW

File: configuration/config.go [hunk 1]
```diff
@@ -273,10 +269,7 @@ func InitCLI(runFunc, rmFunc, cgroupFunc, enforceFunc, daemonFunc func(*Configur
 	cmdDaemon.Flags().StringSlice("target-networks", nil, "The target networks that Trireme should apply authentication")
 	cmdDaemon.Flags().String("policy", "", "Policy file")
 	fUsePKI = cmdDaemon.Flags().Bool("usePKI", false, "Use PKI for Trireme")
-	cmdDaemon.Flags().Bool("hybrid", false, "Hybrid mode of deployment (docker+processes)")
-	fLocal = cmdDaemon.Flags().Bool("local", false, "Local mode of deployment")
 	// TODO: looks superfluous
-	cmdDaemon.Flags().Bool("remote", false, "Remote mode of deployment")
```
Comment (mheese) [inline configuration/config.go]: @dstiliadis thanks for clarifying the flags usage here now ... you can also remove my TODO line

**Comment by mheese** [comment]
@dstiliadis I remember from our last discussion that you said that the new Trireme version is not going to require this state information anymore - even for Linux processes, this is why I picked this scenario for testing :)

If I understand it correctly, with docker we don't need to persist because the information will basically always be retrieved from the docker daemon. So for the sake of the example, we should show that for Linux processes, one needs to persist this state information and make it obvious that this is only required for Linux processes and not docker containers.

Persisting to a file is the right approach, but persisting to the config file is definitely the wrong approach. We should persist to a file under `/var/run/trireme-example/` to correctly follow the FHS: it is run-time variable data that does not need to survive a reboot.

All in all, this PR is extremely valuable to me: it really shows the new Trireme in action and helps me to understand what is going on (as I'm not that familiar with the code-base yet). And after all, this is exactly the purpose of this example repository.

**Comment by dstiliadis** [comment]
I can start a different PR with the persistence example. Persistence is always tricky, even for an example. I will edit the README to clarify that persistence is not supported for the example and Linux processes and we can change it later.  https://github.com/aporeto-inc/trireme-example/issues/40

**Comment by mheese** [comment]
@dstiliadis sounds good to me. It makes it clear that this won't work right now and we also have a plan for it going forward. I like it 👍

**Pull request merged**